test/e2e/mcd_test.go

```diff
@@ -425,6 +425,39 @@ func TestReconcileAfterBadMC(t *testing.T) {
 	}
 }
 
+func TestFullyDeleteMCShippedFiles(t *testing.T) {
+	cs := framework.NewClientSet("")
+	workerOldMcFirst := getMcName(t, cs, "worker")
+	mcHostFileFirst := createMCToAddFile("modify-host-file-first", "/etc/something", "mco-test", "root")
+	_, err := cs.MachineConfigs().Create(mcHostFileFirst)
+	require.Nil(t, err)
+	renderedConfig, err := waitForRenderedConfig(t, cs, "worker", mcHostFileFirst.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	workerOldMcSecond := getMcName(t, cs, "worker")
+	mcHostFileSecond := createMCToAddFile("modify-host-file-second", "/etc/something", "mco-test-2", "root")
+	_, err = cs.MachineConfigs().Create(mcHostFileSecond)
+	require.Nil(t, err)
+	renderedConfig, err = waitForRenderedConfig(t, cs, "worker", mcHostFileSecond.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	// now delete the MCs file, which must result in the file disappearing from the system
+	require.Nil(t, cs.MachineConfigs().Delete(mcHostFileSecond.Name, &metav1.DeleteOptions{}))
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", workerOldMcSecond))
```

Comment:
Would be good to check here that the file still exists and has the contents dictated by the first MC.

Reply:
that's the usual MCD workflow tho - here's is just diffing MC1<->MC2 so it's "applying" MC1 and we expect that to work (and tested in other tests as well)

Reply:
i.e. there's not "removal of stale files in this scenario" so there's little value in (re) testing that here